### ISSUE TITLE
deepsea-0.9.9 throws "IOLoop is already running" exception in stage.0

### ISSUE THREAD

**Issue by craiggardner** [SUSE/DeepSea#1504]
### Description of Issue/Question

I have deepsea-0.9.9 installed on SLES15 SP1.  salt version is 2018.3.0.  When running deepsea stage.0, all minions throw the following exception:

"""
                       ID: switch kernel
                  Function: module.run
                      Name: kernel.replace
                    Result: False
                   Comment: Module function kernel.replace threw an exception. Exception: IOLoop is already running
                   Started: 14:12:40.008724
                  Duration: 531.024 ms
                   Changes:   
                Name: packagemanager.up - Function: module.run - Result: Changed Started: - 14:12:40.540112 Duration: 2720.922 ms
                Name: packagemanager.up - Function: module.run - Result: Changed Started: - 14:12:43.277655 Duration: 2822.681 ms
"""

Wanting to figure this out without having to create an issue here, I googled "IOLoop is already running" for more salt information, and I came up short.

**Comment by craiggardner** [comment]
I am observing the same behavior and error with deepsea 0.9.10

**Comment by jschmid1** [comment]
Fixed with newer salt versions

**Comment by johanneskastl** [comment]
Could you elaborate a bit on which "newer salt versions" this may be?

I am hitting this on SLES 15 SP1 with `salt-2019.2.0-6.14.1.x86_64`...

**Comment by jschmid1** [comment]
> Could you elaborate a bit on which "newer salt versions" this may be?
> 
> I am hitting this on SLES 15 SP1 with `salt-2019.2.0-6.14.1.x86_64` (but could work around it...).

Could you try to update on the latest version in your channels? I'm not entirely sure which version fixed this issue, but we're not seeing this anymore with the current testing environment

**Comment by johanneskastl** [comment]
I already have the latest updates:

```
# zypper se -sx salt
Refreshing service 'Basesystem_Module_15_SP1_x86_64'.
Refreshing service 'Containers_Module_15_SP1_x86_64'.
Refreshing service 'SUSE_Enterprise_Storage_6_x86_64'.
Refreshing service 'SUSE_Linux_Enterprise_Server_15_SP1_x86_64'.
Refreshing service 'Server_Applications_Module_15_SP1_x86_64'.
Loading repository data...
Reading installed packages...

S | Name | Type       | Version         | Arch   | Repository                                  
--+------+------------+-----------------+--------+---------------------------------------------
i | salt | package    | 2019.2.0-6.14.1 | x86_64 | SLE-Module-Basesystem15-SP1-Updates         
v | salt | package    | 2019.2.0-6.11.1 | x86_64 | SLE-Module-Basesystem15-SP1-Updates         
v | salt | package    | 2019.2.0-6.8.1  | x86_64 | SLE-Module-Basesystem15-SP1-Updates         
v | salt | package    | 2019.2.0-6.3.5  | x86_64 | SLE-Module-Basesystem15-SP1-Updates         
v | salt | package    | 2019.2.0-4.4    | x86_64 | SLE-Module-Basesystem15-SP1-Pool            
  | salt | srcpackage | 2019.2.0-6.14.1 | noarch | SLE-Module-Basesystem15-SP1-Updates         
  | salt | srcpackage | 2019.2.0-6.14.1 | noarch | SLE-Module-Server-Applications15-SP1-Updates
  | salt | srcpackage | 2019.2.0-6.11.1 | noarch | SLE-Module-Basesystem15-SP1-Updates         
  | salt | srcpackage | 2019.2.0-6.11.1 | noarch | SLE-Module-Server-Applications15-SP1-Updates
  | salt | srcpackage | 2019.2.0-6.8.1  | noarch | SLE-Module-Basesystem15-SP1-Updates         
  | salt | srcpackage | 2019.2.0-6.8.1  | noarch | SLE-Module-Server-Applications15-SP1-Updates
  | salt | srcpackage | 2019.2.0-6.3.5  | noarch | SLE-Module-Basesystem15-SP1-Updates         
  | salt | srcpackage | 2019.2.0-6.3.5  | noarch | SLE-Module-Server-Applications15-SP1-Updates
``` 

But as I said, I could workaround the issue.

**Comment by jschmid1** [comment]
I found this: 

https://github.com/SUSE/DeepSea/pull/1740

What's your DeepSea version?

**Comment by johanneskastl** [comment]
Also the latest one available:

```
# zypper se -sx deepsea
Refreshing service 'Basesystem_Module_15_SP1_x86_64'.
Refreshing service 'Containers_Module_15_SP1_x86_64'.
Refreshing service 'SUSE_Enterprise_Storage_6_x86_64'.
Refreshing service 'SUSE_Linux_Enterprise_Server_15_SP1_x86_64'.
Refreshing service 'Server_Applications_Module_15_SP1_x86_64'.
Loading repository data...
Reading installed packages...

S  | Name    | Type       | Version                      | Arch   | Repository                       
---+---------+------------+------------------------------+--------+----------------------------------
i+ | deepsea | package    | 0.9.23+git.0.6a24f24a0-3.3.2 | noarch | SUSE-Enterprise-Storage-6-Updates
v  | deepsea | package    | 0.9.22+git.0.29e84a00e-1.3   | noarch | SUSE-Enterprise-Storage-6-Pool   
   | deepsea | srcpackage | 0.9.23+git.0.6a24f24a0-3.3.2 | noarch | SUSE-Enterprise-Storage-6-Updates
```

**Comment by jschmid1** [comment]
ok, the patch for this will be included in 0.9.25 which will be released shortly.

**Comment by johanneskastl** [comment]
Thanks for the information.
